### PR TITLE
Gumbolt

### PR DESCRIPTION
**Pull request by abhishekabhishek**
- FIxed DVAE++ implementation
- Gumbolt initial implementation - w/out evaluation - only training possible

### CODE REVIEW

File: models/autoencoders/autoencoderbase.py [hunk 1]
```diff
@@ -38,6 +38,9 @@ def __init__(self, flat_input_size, train_ds_mean, activation_fct, cfg, **kwargs
 
         # the main configuration namespace returned by configaro
         self._config=cfg
+
```
Comment (ezeeEric) [inline models/autoencoders/autoencoderbase.py:41]: Hey @abhishekabhishek, it's a minor thing, but not all models inheriting from autoencoderbase need this variable. So I would prefer to keep it in the Discrete VAE models - afaik the `DiVAEpp` and `Gumbolt` models inherit from `DiVAE`, so we could put it there.

File: models/networks/hierarchicalEncoder.py [hunk 2]
```diff
@@ -11,10 +11,14 @@
 from utils.dists.distributions import SpikeAndExponentialSmoother
 from utils.dists.MixtureExp import MixtureExp
 from utils.dists.MixtureExpMod import MixtureExpMod
+from utils.dists.GumbelMod import GumbelMod
 
 _SMOOTHER_DICT = {"SpikeExp" : SpikeAndExponentialSmoother, 
-                  "MixtureExp" : MixtureExp}
-_SMOOTHER_MOD_DICT = {"MixtureExp" : MixtureExpMod, "SpikeExp" : None}
+                  "MixtureExp" : MixtureExp,
+                  "Gumbel" : None}
```
Comment (ezeeEric) [inline models/networks/hierarchicalEncoder.py:18]: Is it Gumbel or GumBolt or both? :D Confused!

Comment (abhishekabhishek) [reply models/networks/hierarchicalEncoder.py:18]: Gumbel is the smoothing procedure of adding Gumbel-Softmax noise and Gumbolt is the application of this procedure to Boltzmann machines.

File: models/autoencoders/autoencoderbase.py (hunk 1 above)
Comment (abhishekabhishek) [reply models/autoencoders/autoencoderbase.py:41]: i believe it already existed in autoencoderbase. i did not add this here.

Comment (ezeeEric) [reply models/autoencoders/autoencoderbase.py:41]: Thanks!

Comment (ezeeEric) [reply models/autoencoders/autoencoderbase.py:41]: @abhishekabhishek I removed this in c4fb11d3ee36888638bebd6071ed367888e8794e and replaced it with the corresponding member var from the `DiVAE` model.

**Review by ezeeEric — APPROVED** [review]

**Pull request merged**